src/webui/webapplication.cpp

```diff
@@ -735,6 +737,28 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocation = request().posts["location"].trimmed();
+
+    // check if location valid
+    if(newLocation.isEmpty() || !QDir(newLocation).exists()) return;
+
+    foreach (const QString &hash, hashes) {
+        BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+        if (torrent) {
+            const QString oldLocation = torrent->savePath();
```

Comment:
multiple hashes implies multiple torrents are being moved to new location. how does the directory structure look like after movement? should the method not take multiple new locations as argument, one for each torrent?

Reply:
In case of multiple torrents, all will be moved to same location. Although that won't be used in our case, but the intention was to impl the [documented api](https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-Documentation#set-torrent-location) to avoid confusion.